frontend/device/remarkable/device.lua

```diff
@@ -60,18 +78,35 @@ local adjustTouchEvt = function(self, ev)
     end
 end
 
+
 function Remarkable:init()
     self.screen = require("ffi/framebuffer_mxcfb"):new{device = self, debug = logger.dbg}
-    self.powerd = require("device/remarkable/powerd"):new{device = self}
+    self.powerd = require("device/remarkable/powerd"):new{
+        device = self,
+        capacity_file = self.battery_path,
+        status_file = self.status_path,
+    }
     self.input = require("device/input"):new{
         device = self,
         event_map = require("device/remarkable/event_map"),
     }
 
-    self.input.open("/dev/input/event0") -- Wacom
-    self.input.open("/dev/input/event1") -- Touchscreen
-    self.input.open("/dev/input/event2") -- Buttons
-    self.input:registerEventAdjustHook(adjustTouchEvt)
+    self.input.open(self.input_wacom) -- Wacom
+    self.input.open(self.input_ts) -- Touchscreen
+    self.input.open(self.input_buttons) -- Buttons
+
+    self.input:registerEventAdjustHook(adjustAbsEvt)
+
+    if self.invertX() then
+        self.input:registerEventAdjustHook(self.input.adjustTouchMirrorX, self.mt_width)
+    end
+
+    local mt_scale_x = self.mt_width / screen_width
+    local mt_scale_y = self.mt_height / screen_height
+
+    self.input:registerEventAdjustHook(self.input.adjustTouchMirrorY, self.mt_height)
```

Comment:
this affects the pen as well

Reply:
That's quite likely, and possibly hard to deal with unless there's a sure-fire way to disciminate the two solely based on the events themselves.

(The input subsystem basically chucks everything in one feed, so you can't really know where the event came from, IIRC).

Reply:
@NiLuJe: give it a review on my suggestion in https://github.com/ddvk/koreader/pull/2, yay or nay?

Reply:
@pazos: I'd keep the `local TimeVal = require('ui/timeval')`, but, sure, why not. (Note that the fact this only handles the MT events, unlike the global hook, might be enough to skip the pen, since I hope the pen doesn't actually generate MT events ^^).

Reply:
i think i'll just revert the old adjust handler and have a version for the rm2

Reply:
Yep, that works, too ;).